### PR TITLE
Additional Flags in OMPL Settings + Default State in Scene YAML

### PR DESCRIPTION
**Pull request by ChamzasKonstantinos**
Added simplification/interpolation flags in OMPL settings. 
Added the default state in a scene when loaded from YAML  (removes found empty joint_state message)

### CODE REVIEW

File: robowflex_library/include/robowflex_library/planning.h [hunk 1]
```diff
@@ -312,6 +312,8 @@ namespace robowflex
             int max_state_sampling_attempts;        ///< Maximum number of attempts to sample a new state.
             int minimum_waypoint_count;             ///< Minimum number of waypoints in output path.
             bool simplify_solutions;                ///< Whether or not planner should simplify solutions.
+            bool hybridize_solutions;               ///< Whether or not planner should hybridize solutions.
+            bool interpolate_solutions;             ///< Whether or not planner should interpolate solutions.
```
Comment (zkingston) [inline robowflex_library/include/robowflex_library/planning.h:316]: Shouldn't these also be set as parameters on the server? Does the OMPL plugin understand those?

File: robowflex_library/src/scene.cpp [hunk 2]
```diff
@@ -562,6 +563,9 @@ bool Scene::fromYAMLFile(const std::string &file)
         return false;
 
     fixCollisionObjectFrame(msg);
+    // Add robot_state if loaded scene does not contain one.
```
Comment (zkingston) [inline robowflex_library/src/scene.cpp:566]: Add space above.

File: robowflex_ompl/include/robowflex_ompl/ompl_interface.h [hunk 3]
```diff
@@ -91,6 +91,8 @@ namespace robowflex
 
             std::unique_ptr<ompl_interface::OMPLInterface> interface_{nullptr};  ///< Planning interface.
             std::vector<std::string> configs_;                                   ///< Planning configurations.
+            bool hybridize_{true};    ///< Whether or not planner should hybridize solutions.
```
Comment (zkingston) [inline robowflex_ompl/include/robowflex_ompl/ompl_interface.h:94]: Do these need default values if they are always overridden by the values in the settings?

**Review by zkingston — APPROVED** [review]
Some comments, but looks fine.

File: robowflex_ompl/include/robowflex_ompl/ompl_interface.h (hunk 3 above)
Comment (ChamzasKonstantinos) [reply robowflex_ompl/include/robowflex_ompl/ompl_interface.h:94]: No it does not, you can only modify them through the [ModelBasedPlanningContext](https://github.com/ros-planning/moveit/blob/master/moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/model_based_planning_context.h#L267)
and not through the [ompl_interface](https://github.com/ros-planning/moveit/blob/master/moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/ompl_interface.h#L122) although we should change MoveIt at some point to do so

File: robowflex_ompl/src/ompl_interface.cpp [hunk 4]
```diff
@@ -112,6 +114,8 @@ void OMPL::OMPLInterfacePlanner::refreshContext(const SceneConstPtr &scene,
         return;
     }
 
+    context_->setInterpolation(interpolate_);
+    context_->setHybridize(hybridize_);
```
Comment (zkingston) [inline robowflex_ompl/src/ompl_interface.cpp:118]: Looks like these functions only got added in melodic, you should add a header blocker around them.

Comment (ChamzasKonstantinos) [reply robowflex_ompl/src/ompl_interface.cpp:118]: Yeah, I was the one who added them actually

File: robowflex_library/include/robowflex_library/planning.h (hunk 1 above)
Comment (ChamzasKonstantinos) [reply robowflex_library/include/robowflex_library/planning.h:316]: No it does not, you can only modify them through the [ModelBasedPlanningContext](https://github.com/ros-planning/moveit/blob/master/moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/model_based_planning_context.h#L267)
and not through the [ompl_interface](https://github.com/ros-planning/moveit/blob/master/moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/ompl_interface.h#L122) although we should change MoveIt at some point to do so

**Pull request merged**